dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/GroupRestRepository.java

```diff
@@ -149,4 +172,77 @@ protected void patch(Context context, HttpServletRequest request, String apiCate
     public Class<GroupRest> getDomainClass() {
         return GroupRest.class;
     }
+
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected void delete(Context context, UUID uuid) throws AuthorizeException {
+        Group group = null;
+        try {
+            group = gs.find(context, uuid);
+            if (group == null) {
+                throw new ResourceNotFoundException(
+                        GroupRest.CATEGORY + "." + GroupRest.NAME
+                                + " with id: " + uuid + " not found"
+                );
+            }
+            if (isDeleteUnprocessable(context, group)) {
+                throw new UnprocessableEntityException("This group cannot be deleted");
+            }
+            gs.delete(context, group);
+        } catch (SQLException | IOException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    private boolean isDeleteUnprocessable(Context context, Group group) {
+        try {
+            return group.isPermanent() || gs.getParentObject(context, group) != null;
```

Comment:
There are separate endpoints for that purpose: https://github.com/DSpace/Rest7Contract/pull/105

It would be best that groups are deleted using those endpoints, if we delete from the general group endpoint you could be deleting a group that is attached a community/collection & this would inadvertently delete the link. We can always update the error description to state which object this group is linked to ?

Reply:
@KevinVdV : good point. My mistake, as I didn't recall this was being managed as a separate endpoint. In that case, I agree with you that we should only allow deletion of Community/Collection Groups via the communities and collections endpoints.  Thanks for clarifying this!